### ISSUE TITLE
Caching and versioning of data

### ISSUE THREAD

**Issue by tlhunter** [tlhunter/node-grille#9]
I was going to "leave this up to the reader", but it's vital enough that it should really be in the module.

**Comment by tlhunter** [comment]
Allow the passing in of a storage class to the Registry which requires a few methods:

```javascript
save(version, data, function(err} {});
getCurrentVersion(function(err, version) {});
load(version, function(err, data) {});
loadCurrentVersion(function(err, data) {});
list(function(err, versions) {});
```

Storage will need to track the "current version", and some sort of key/value pairing of versions to data.

Also provide a Redis based example.

**Comment by tlhunter** [comment]
Fixed in 2249195a875aae769f5ec172bd91d8e5fcecd838
